kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/LayoutRunner.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow.ui
+
+import android.content.Context
+import android.support.annotation.LayoutRes
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import kotlin.reflect.KClass
+
+/**
+ * An object that handles [View.showValue] calls for a view inflated
+ * from a layout resource in response to [ViewRegistry.buildView].
+ * (Use [BuilderBinding] if you want to build views from code rather than
+ * layouts.)
+ *
+ * Typical usage is to have a [LayoutRunner]'s `companion object` implement
+ * [ViewBinding] by delegating to [LayoutRunner.bind], specifying the layout resource
+ * it expects to drive.
+ *
+ *   class HelloLayoutRunner(view: View) : LayoutRunner<Rendering> {
+ *     private val messageView: TextView = view.findViewById(R.id.hello_message)
+ *
+ *     override fun showValue(value: Rendering) {
+ *       messageView.text = value.message
+ *       messageView.setOnClickListener { value.onClick(Unit) }
+ *     }
+ *
+ *     companion object : ViewBinding<Rendering> by bind(
+ *         R.layout.hello_goodbye_layout, ::HelloLayoutRunner
+ *     )
+ *   }
+ *
+ * This pattern allows us to assemble a [ViewRegistry] out of the
+ * [LayoutRunner] classes themselves.
+ *
+ *    val TicTacToeViewBuilders = ViewRegistry(
+ *        NewGameLayoutRunner, GamePlayLayoutRunner, GameOverLayoutRunner
+ *    )
+ *
+ * Also note that two flavors of [contructor][LayoutRunner.Binding.runnerConstructor]
+ * are accepted by [bind]. Every [LayoutRunner] constructor must accept an [View].
+ * Optionally, they can also have a second [ViewRegistry] argument, to allow
+ * nested values to be displayed via nested calls to [ViewRegistry.buildView].
+*/
+@ExperimentalWorkflowUi
+interface LayoutRunner<ValueT : Any> {
+  fun showValue(value: ValueT)
+
+  class Binding<ValueT : Any>
+  constructor(
+    override val type: KClass<ValueT>,
+    @LayoutRes private val layoutId: Int,
+    private val runnerConstructor: (View, ViewRegistry) -> LayoutRunner<ValueT>
+  ) : ViewBinding<ValueT> {
```

Comment:
Are we intentionally calling this ValueT vs RenderingT? Any reason to prefer showValue() vs render()?

Reply:
I kept going back and forth on that. If your intuition is leading the same way, I'll go through and make this `RenderingT` and `render()`.

Reply:
Oh yeah, so much better. And `ViewShowRendering.kt`, `View.bindShowRendering`, `initialRendering`, etc.